### ISSUE TITLE
Possible sorting bug in out-of-the-box Astro blog example

### ISSUE THREAD

**Issue by jeff-r-koyaltech** [withastro/astro#11586]
### Astro Info

```block
Astro                    v4.11.5
Node                     v20.10.0
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
                         @astrojs/sitemap
```

### If this issue only occurs in one browser, which browser is a problem?

Any browser

### Describe the Bug

Root cause can be seen here:
https://github.com/search?q=repo%3Awithastro%2Fastro%20data.pubDate.valueOf()&type=code

In both the example site and the site generated by `npm create astro@latest my-new-shiny-site`, the sorting puts the oldest blog posts at the top.

(This would be a trivial fix to just do b minus a instead of a minus b...if someone from Astro wants to confirm they'd welcome a quick PR to fix those 3 lines, I'd be happy to submit it tomorrow or Friday and contribute!)

### What's the expected result?

I feel like most real world use cases would want the newest blog posts at the top of the page.

### Link to Minimal Reproducible Example

Run `npm create astro@latest my-new-shiny-site`

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by jeff-r-koyaltech** [comment]
I got started in a fork:
https://github.com/withastro/astro/pull/11587

**Comment by ascorbic** [comment]
Closed in #11587. Thanks!
